src/hal/i_components/stats.icomp

```diff
@@ -5,9 +5,9 @@ pin in bit enable "";
 pin out float mean;
 pin out float variance;
 pin out u32 n;
-variable int prev_enable;
-variable float m2;
-variable float delta;
+variable hal_s32_t prev_enable;
+variable hal_float_t m2;
+variable hal_float_t delta;
 function _;
 license "GPL";
 ;;
```

Comment:
implicit typecast of n in line 26 and 28 should be explicitly cast to hal_float_t